synapse/storage/databases/main/events_bg_updates.py

```diff
@@ -1310,12 +1310,9 @@ def process(txn: Cursor) -> None:
 
         # ANALYZE the new column to build stats on it, to encourage PostgreSQL to use the
         # indexes on it.
-        # We need to pass execute a dummy function to handle the txn's result otherwise
-        # it tries to call fetchall() on it and fails because there's no result to fetch.
-        await self.db_pool.execute(
+        await self.db_pool.runInteraction(
```

Comment:
Is the point that we should use `runInteraction` when we don't expect---or don't care---about the result set?

Reply:
and the only place where we used a decoder before was where we were trying to ignore the result set!?!

Reply:
> Is the point that we should use `runInteraction` when we don't expect---or don't care---about the result set?

`execute(...)` automatically calls `fetchall()` (or iterates the `txn` if you use `cursor_to_dict`. You could do hacky things like this to *not* iterate the `txn` and just leave it, but...I think this is as clear.

> and the only place where we used a decoder before was where we were trying to ignore the result set!?!

Or we passed `cursor_to_dict` in many many places, which are all gone.